mne/source_estimate.py

```diff
@@ -1691,6 +1770,121 @@ def get_peak(self, tmin=None, tmax=None, mode='abs',
                 time_idx if time_as_index else self.times[time_idx])
 
 
+class VectorSourceEstimate(SourceEstimate):
+    """Container for full MNE surface estimate, including dipole directions
+
+    Parameters
+    ----------
+    data : array of shape (n_dipoles, 3, n_times)
+        The data in source space. Each dipole contains three vectors that
+        denote the dipole strength in X, Y and Z directions over time.
+    vertices : array | list of two arrays
+        Vertex numbers corresponding to the data.
+    tmin : float
+        Time point of the first sample in data.
+    tstep : float
+        Time step between successive samples in data.
+    subject : str | None
+        The subject name. While not necessary, it is safer to set the
+        subject parameter to avoid analysis errors.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Attributes
+    ----------
+    subject : str | None
+        The subject name.
+    times : array of shape (n_times,)
+        The time vector.
+    shape : tuple
+        The shape of the data. A tuple of int (n_dipoles, n_times).
+    Notes
+    -----
+    .. versionadded:: 0.13
+    """
+    @verbose
+    def __init__(self, data, vertices=None, tmin=None,
+                 tstep=None, subject=None, verbose=None):
+        SourceEstimate.__init__(self, data, vertices=vertices, tmin=tmin,
+                                tstep=tstep, subject=subject, verbose=verbose)
+
+    @verbose
+    def save(self, fname, ftype='h5', verbose=None):
+        """Save the full source estimate to an HDF5 file
+
+        Parameters
+        ----------
+        fname : string
+            The stem of the file name, should end in '-stc.h5'.
+        ftype : string
+            File format to use. Currently, the only allowed values is "h5".
+        verbose : bool, str, int, or None
+            If not None, override default verbose level (see mne.verbose).
+            Defaults to self.verbose.
+        """
+        if ftype != 'h5':
+            raise ValueError('VectorSourceEstimate objects can only be '
+                             'written as HDF5 files.')
+
+        if not fname.endswith('-stc.h5'):
+            fname += '-stc.h5'
```

Comment:
IIRC hdf5 files have a few compatible filename endings. I also don't think we should force `-stc`, just recommend it (as we do with various flavors of FIF), since it should be possible to tell from the contents of the file what type of object to reconstruct